File: astah/jude_api/usecaselistup.rb
require 'java'

if ARGV.empty?
	puts ">jruby #{__FILE__} [jude file]"
	exit
end

module Jude
    include_package "com.change_vision.jude.api.inf.project"
    include_package "com.change_vision.jude.api.inf.model"
end

pro = Jude::ProjectAccessorFactory.projectAccessor
pro.open ARGV[0]

actors = []
usecases = {}

pro.project.ownedElements.each do |oe|
	actors.push(oe) if oe.stereotypes.include?("actor")
	usecases.store(oe.name, oe) if oe.java_kind_of? Jude::IUseCase
end

actors.each do |actor|
	puts "actor : #{actor.name}"

	actor.attributes.each do |ass|

		if ass.association
			uc = usecases[ass.typeExpression]
			puts "usecase : #{uc.name}"
		end
	end

end

pro.close

File: astah/jude_api/usecase_listup.scala

import com.change_vision.jude.api.inf.project._
import com.change_vision.jude.api.inf.model._

case class Model(name: String)
case class UseCase(usecase: IUseCase) extends Model(usecase.getName())
case class Actor(actor: INamedElement) extends Model(actor.getName())

val proj = ProjectAccessorFactory.getProjectAccessor()
proj.open(args(0))

val list = proj.getProject.getOwnedElements.flatMap {
	case u: IUseCase => List(UseCase(u))
	case a if a.getStereotypes.contains("actor") => List(Actor(a))
	case _ => None
}

list.foreach {n =>
	println(n.getClass.getSimpleName + "\t" + n.name)
}

proj.close
